build/preprocessing.py

```diff
@@ -0,0 +1,55 @@
+#! python
+
+import yaml
+import sys
+
+if __name__ == "__main__":
+
+  print("Running preprocessing for common templates")
+
+  rules = None
+  patchPath = sys.argv[1] if len(sys.argv) > 1 else "patch.yaml"
+  with open(patchPath, 'r') as stream:
+    try:
+      patch = yaml.load(stream, Loader=yaml.BaseLoader)
+
+      if patch == None: 
+        print("Nothing in patch file")
+        sys.exit(0)
+      
+      rules = patch.get("rules")
+      for rule in rules:
+        rule["matchLabel"] = rule.get("matchLabel").split(": ")
+        rule["addAnnotation"] = rule.get("addAnnotation").split(": ")
+    except yaml.YAMLError as exc:
+      print(exc)
+      sys.exit(1)
+
+  commonTemplates = None
+  commonTemplatesPath = sys.argv[2] if len(sys.argv) > 2 else "commonTemplates.yaml"
+  with open(commonTemplatesPath, 'r') as stream:
+    try:
+      commonTemplates = list(yaml.safe_load_all(stream))
+      if len(commonTemplates) == 0: 
+        print("Nothing in common templates file")
+        sys.exit(0)
+
+    except yaml.YAMLError as exc:
+      print(exc)
+      sys.exit(1)
+
+  for template in commonTemplates:
+    metadata = template.get("metadata", {})
+    annotations = metadata.get("annotations",{})
+    labels = metadata.get("labels", {})
+    for rule in rules:
+      key = rule.get("matchLabel")[0]
+      value = rule.get("matchLabel")[1]
+      if labels.get(key) == value:
+        annotationKey = rule.get("addAnnotation")[0]
```

Comment:
Please add a debug print here that prints the template name and the added annotation